### ISSUE TITLE
Restart on EADDRINUSE

### ISSUE THREAD

**Issue by tshemsedinov** [metarhia/impress#1890]
### Describe the problem

```
uncaughtException: Error: listen EADDRINUSE: address already in use 0.0.0.0:8000
  Server.setupListenHandle [as _listen2] (node:net:1463:16)
```

### Describe the solution

Port is in use by another server, so we can:
- wait for timeout
- try 3 times
- exit 1

**Comment by MarhiievHE** [comment]
@tshemsedinov  It looks like the HTTP server is not returning this error via the error listener: https://github.com/metarhia/metacom/blob/c8565a96eb0b77a536b15f643f4a727321478972/lib/server.js#L213. Will it be normal to catch this error in this way in worker.js:

```js
    process.on('uncaughtException', function(err) {
      if (err.code === 'EADDRINUSE') {
        console.warn(`Address in use: ${options.host}:${port}, retry...`);
        const timer = setTimeout(() => {
          console.log({tryToStartCounter});
          tryToStartCounter++;
          if (tryToStartCounter > 3) process.kill(process.pid, 'SIGINT');
          application.server = new metarhia.metacom.Server(application, options);
        },options.timeouts.bind);
      }
    });
```

**Comment by MarhiievHE** [comment]
@tshemsedinov It looks like the HTTP server is not returning this error via the error listener. Will it be ok to catch this error in this way in worker.js, or do we need to find a more elegant solution?:
```js
    process.on('uncaughtException', function(err) {
      if (err.code === 'EADDRINUSE') {
        console.warn(`Address in use: ${options.host}:${port}, retry...`);
        const timer = setTimeout(() => {
          console.log({tryToStartCounter});
          tryToStartCounter++;
          if (tryToStartCounter > 3) process.kill(process.pid, 'SIGINT');
          application.server = new metarhia.metacom.Server(application, options);
        },options.timeouts.bind);
      }
    });
```

**Comment by MarhiievHE** [comment]
> @tshemsedinov It looks like the HTTP server is not returning this error via the error listener. Will it be ok to catch this error in this way in worker.js, or do we need to find a more elegant solution?:
> 
> ```js
>     process.on('uncaughtException', function(err) {
>       if (err.code === 'EADDRINUSE') {
>         console.warn(`Address in use: ${options.host}:${port}, retry...`);
>         setTimeout(() => {
>           tryToStartCounter++;
>           if (tryToStartCounter > 3) process.kill(process.pid, 'SIGINT');
>           application.server = new metarhia.metacom.Server(application, options);
>         },options.timeouts.bind);
>       }
>     });
> ```
> 
> https://github.com/MarhiievHE/impress/blob/594c068dc9e97a9c3734dcc44251e32814aa5492/lib/worker.js#L86-L102

I was wrong, that wasn't the point. The problem was that a wsServer was created in addition to the httpServer, and the error occurred in the wsServer, so the httpServer error listener did not catch it.

**Comment by MarhiievHE** [comment]
It makes three tries to open the server on the specified port with bind timeout and then sends SIGTERM to end with end code 1.
<img width="581" alt="image" src="https://github.com/metarhia/impress/assets/12165613/2de9324a-bd9b-4af4-979f-1a489b1929e6">

**Comment by tshemsedinov** [comment]
Fixed in https://github.com/metarhia/impress/pull/1908 and published 3.0.7
